packages/components/src/utils/events.ts

```diff
@@ -30,22 +30,31 @@ function mergeEvent< TEvent extends Event >(
  */
 export function mergeEventHandlers<
 	TEvent extends Event,
-	TLeft extends Record< string, EventHandler< TEvent > >
->(
-	handlers: TLeft,
-	extraHandlers: Record< string, EventHandler< TEvent > >
-): TLeft {
-	const mergedHandlers: TLeft = { ...handlers };
+	TLeft extends Record< string, EventHandler< TEvent > >,
+	TRight extends Record< string, EventHandler< TEvent > >
+>( handlers: TLeft, extraHandlers: TRight ): TLeft & TRight {
```

Comment:
Would `Record< keyof TLeft & keyof TRight, EventHandler< TEvent > >` be a more correct return type?

Reply:
No, `typeof TLeft` would be `Record<string, EventHandler<TEvent>>` as would `typeof TRight`.

Reply:
That's true, but in my previous comment I used `keyof` (as opposed to `typeof`)

Reply:
Oh I see. No unfortunately `keyof Record<string, T>` is just `string`.

Reply:
For the record I'm mirroring the signature of `assign` from `lodash` which I think is a battle tested and accurate type signature.

Reply:
Thank you for the explanation!